server/text_generation_server/server.py

```diff
@@ -29,6 +29,11 @@ def __init__(self, model: Model, cache: Cache, server_urls: List[str]):
     async def Info(self, request, context):
         return self.model.info
 
+    async def Health(self, request, context):
+        if self.model.device.type == "cuda":
+            torch.zeros((2, 2)).to(device=f"cuda:{os.environ['RANK']}")
```

Comment:
You can .cuda() directly since the default device is set already to the correct rank.

Reply:
Done.